### ISSUE TITLE
Scroll to a specific line in codemirror

### ISSUE THREAD

**Issue by GevinHasmitha** [uiwjs/react-codemirror#617]
I'm adding a classname to a specific line in the codemirror using classnameExt function provided by uiw codemirror. I have access to the classname and the line number of the specific line. I want to scroll to that specific line on a button click.

`<CodeMirror
      id="codemirrorId"
      value={value}
      height={height}
      theme={darkMode ? aura : xcodeLight}
      extensions={extensions}
      onChange={onChange}
      readOnly={readOnly}
      color="text.primary"
    />`

This is my codemirror component, how can i scroll to the specific line which has my classname?

**Comment by jaywcjlove** [comment]
@GevinHasmitha  I don't understand what you mean. Do you want to scroll to the specified line?

**Comment by GevinHasmitha** [comment]
@jaywcjlove Yes, I want the editor to scroll to the specific line where my added classname is present  when i click on a button. (Im adding the classname using the classnameExt extension)

**Comment by jaywcjlove** [comment]
<img width="1061" alt="Screenshot 2024-01-17 at 13 24 39" src="https://github.com/uiwjs/react-codemirror/assets/1680273/7ebcc0ea-8870-4776-9309-0705befd405f">

You need to get the dom node of `div.cm-scroller` to control the scroll bar. 

Then get the height of the row and position it.

**Comment by jaywcjlove** [comment]
https://github.com/uiwjs/react-codemirror/blob/bf3b862923d0cb04ccf4bb9da0791bdc7fd6d29b/extensions/events/src/index.ts#L18-L48

**Comment by GevinHasmitha** [comment]
` const scroller = document.getElementsByClassName("cm-scroller")[0];
    if (scroller) {
      const totalLines = 200     /* total number of lines */
      const totalHeight = scroller.scrollHeight;
      const singleLineHeight = totalHeight / totalLines;
  
      scroller.scrollTop = (singleLineHeight * x);   /*x is the line number to scroll to*/
    }`

**Comment by flatsiedatsie** [comment]
> ```
>  const scroller = document.getElementsByClassName("cm-scroller")[0];
>     if (scroller) {
>       const totalLines = 200     /* total number of lines */
>       const totalHeight = scroller.scrollHeight;
>       const singleLineHeight = totalHeight / totalLines;
>   
>       scroller.scrollTop = (singleLineHeight * x);   /*x is the line number to scroll to*/
>     }
> ```

That only works if the lines are of equal height. When using CodeMirror as a text-editor that might not be the case.
